HIP-6.md

```diff
@@ -0,0 +1,147 @@
+ * signed checkpoints (e.g. AWS S3)
+ * @param _storageMetadata Information encoding the location of signed
+ * checkpoints, specific to the storage modaility (e.g. S3 bucket URL)
+ * @return The digest to EIP-191 sign
+ */
+function getDigestToSign(
+    uint32 _domain,
+    bytes32 _mailbox,
+    uint8 _storageType,
+    bytes calldata _storageMetadata
+)
+    external
+    pure
+    returns (bytes32)
+{
+    bytes32 _domainHash = keccak256(
+        abi.encodePacked(_domain, _mailbox, "HYPERLANE")
+    );
+    return keccak256(abi.encodePacked(_domainHash, _storageType, _storageMetadata));
+}
+```
+
+#### Storage types
+
+To be forward compatible with alternative storage modalities, the announcement protocol includes a "storage type" parameter.
+
+Upon adoption, the only supported storage type will be AWS S3, which has `storageType == 1`.
+
+For the S3 type, `_storageMetadata` should be the base64-encoded URL of the S3 bucket.
+
+Future storage types may be proposed in future HIPs. The content of `_storageMetadata` is expected to vary by type.
+
+#### ValidatorRegistry
+
+A contract that implements the `IValidatorRegistry` interface should be deployed to each chain.
+
+This contract stores validator registrations for that chain and mailbox contract.
+
+Validators for that domain and mailbox contract should register themselves so that they are discoverable by relayers.
+
+```
+interface IValidatorRegistry {
+    /// @notice Returns the local domain for validator registrations
+    function localDomain() external view returns (uint32);
+
+    /// @notice Returns the mailbox contract for validator registrations
+    function mailbox() external view returns (address);
+
+    /// @notice Returns a list of validators that have registered
+    function validators() external view returns (address[] memory);
+
+    /**
+     * @notice Returns a list of all registrations for all provided validators
+     * @param _validators The list of validators to get registrations for
+     * @return A list of validator addresses, and registered storage types and
+     * metadata. Note that the validators return value will only match
+     * _validators when each validator in _validators has exactly one
+     * registration.
+     */
+    function getValidatorRegistrations(
+        address[] calldata _validators,
+    ) external view returns (
+        address[] memory validators,
+        uint8[] memory storageTypes,
+        bytes[] memory storageMetadata
+    );
+
+   /**
+    * @notice Registers a validator
+    * @param _storageType An enum representing the modality used to store
+    * signed checkpoints (e.g. AWS S3)
+    * @param _storageMetadata Information encoding the location of signed
+    * checkpoints, specific to the storage modaility (e.g. S3 bucket URL)
+    * @param _signature The validator announcement attestation previously
+    * specified in this HIP
+    * @returns True upon success
```

Comment:
Ive never been a huge fan of relying upon bools for success and have generally preferred reverting upon an error occurring. Wondering if there's a motivation behind bool over reverting?

Reply:
No reason other than following a common pattern, I was imagining this function would return true if successful and revert if not